### PR TITLE
context_doc

### PR DESCRIPTION
**Pull request by ChantelDean**
Context object/api doc.  File is in markdown format.  

Related Jira: DI-3037/DOC-137

---
https://github.com/tetrascience/ts-sdk-python/blob/ChantelDean-contextdoc-di-3037-doc-137/context_doc.md

### CODE REVIEW

**Comment by yandryianets** [comment]
@ChantelDean , can we add a Table of Content?

File: context_doc.md [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+|context.protocol_slug|	string|	Protocol slug.|
+|context.protocol_version|	string|	Protocol version.|
+|context.pipeline_config|	dict|	A dictionary from strings to strings, containing the configuration parameters set when configuring the pipeline that invoked this protocol.|
+|context.input_file|	dict|	File pointer that triggered the current workflow.|
+|context.created_at	|string|	Workflow's creation time.ISO timestamp string. Generated by moment().toISOString(). Indicates when the file is created.|
+|context.task_id|	uuid|	ID for the current task (step of a pipeline).|
+|context.task_created_at|	string|	Task's creation time. ISO timestamp string. Generated by moment().toISOString(). Indicates whent the task is created.|
+|context.platform_url|	string|	The URL of the platform (i.e.: https://platform.tetrascience.com).|
+|context.tmp_dir|	string|	Folder for tmp dir. Usually it’s /tmp. If your task script wants to write a file and then call a shell command (like LibreOffice) to process the file, you should put it in context.tmp_dir . Other folders are not guaranteed to be writeable|.
+context.read_file|string|Takes an input file reference, and returns a dictionary with keys.|
```
Comment (yandryianets) [inline context_doc.md:47]: read_file() should be at the same level as other context functions

**Comment by ChantelDean** [comment]
Yes, I'll add it.

File: context_doc.md [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+
+* An example of a task script entry-point function
+* A description of the context API properties and functions
+* A description of Local Context API functions for unit tests.
+* Task Script Entry-Point Function Example
+
+If your task script entry-point function process_file is in file main.py, you should define process_file like this:
+
+```py
+main.py
```
Comment (chunfengd) [inline context_doc.md:13]: Need to comment this like, like this:
```py
# main.py
...
```

File: context_doc.md [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+### Parameter(s)
+|Parameter|Type|Description|
+|---------|----------|---------|
+|file|	dict|	File Pointer|
+|form|	string|	Optional. Indicates the format for the result. By default, the form is set to body, but it can also be file_obj and download.|
+### Examples
+*body*
+
+```py
+Python
```
Comment (chunfengd) [inline context_doc.md:58]: Please remove "Python" from source code

File: context_doc.md (hunk 1 above)
Comment (ChantelDean) [reply context_doc.md]: Change has been made.  Thanks Yury!

File: context_doc.md (hunk 2 above)
Comment (ChantelDean) [reply context_doc.md]: Change has been made, thanks Chunfeng!

File: context_doc.md (hunk 3 above)
Comment (ChantelDean) [reply context_doc.md]: Change has been made.  Thanks Chunfeng.

File: context_doc.md [hunk 4]
```diff
@@ -0,0 +1,485 @@
...
+
+```py
+
+f = context.read_file(input_file_pointer, form='download')
+con = sqlite3.connect(f['download'])
+df = pd.read_sql_query('SELECT * FROM foo', con)
+```
+
+### Return(s)
+Returns an dictionary, as listed below:
```
Comment (sergei-serezha) [inline context_doc.md:109]: a typo, Returns "A" dictionary

File: context_doc.md [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+
+|Parameter|Type|Description|
+|---------|----------|---------|
+|file|	dict|	File pointer|
+
+### Returns
+A string; the filename of the file. Note that you do not need to download the file locally to access the file name.
+
+## context.get_ids
+Use this function to get the Intermediate Data Schema that matches the namespace, slug, and version.
```
Comment (sergei-serezha) [inline context_doc.md:234]: You can append IDS to all Intermediate Data Schema (IDS). 
The more customers see it the more they get used to it.

File: context_doc.md [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+|custom_metadata|	dict of string to string|	(Optional. Default is `{}`.) Metadata to be appended to a file. Keys can only include letters, numbers, spaces, and the symbols `+, -, /, or .`. Values can only include letters, numbers, spaces, and the symbols `+, -, , /, . or ,`.|
+|custom_tags|	list of strings|Optional. (Default: `[]`)  	Tags to be appended to a file. Can only include letters, numbers, spaces, and the symbols `+, -, ., /, or _)`. Limited to 128 characters.|
+|source_type|	string|	Optional. (Default: `None`) Can be used to overwrite the source type S3 metadata value of the resulting document. Validation is performed if a value is passed against the regex `/^[-a-z0-9]+$/`, if no value is passed, we fall back to the default logic (take source type value from the RAW file, or if not found on RAW file, set the value unknown).|
+|labels| list of dict| Optional. (Default: `[]`)  List of dictionaries where each dictionary is in form `{ 'name': <LABEL_NAME>, 'value': <LABEL_VALUE> }`.  Names can only include letters, numbers, spaces, and the symbols` +, -, or _`. Values can only include letters, numbers, spaces, and the symbols `+, -, _, /, . or ,`.
```
Comment (chunfengd) [inline context_doc.md:134]: label name can include `.`

File: context_doc.md [hunk 7]
```diff
@@ -0,0 +1,485 @@
+# Context API
```
Comment (VladimirShteyn) [inline context_doc.md:1]: Should this file be in a top-level `docs` folder? That seems to be standard practice.

**Comment by spinwang** [comment]
@ChantelDean Considering we are likely moving towards Sphinx, I will close this PR for now. in case we need, we can bring this back. thx.

**Pull request closed without merging**